specs/language/expressions.tex

```diff
@@ -160,3 +160,86 @@
+or \textit{input/output parameters} as denoted in the called function's
+declaration (\ref{Decl.Function}).
+
+\p \textit{Input parameters} are passed by-value into a function. If an argument
+to an \textit{input parameter} is of constant-sized array type, the array is
+copied to a temporary and the temporary value is converted to an address via
+array-to-pointer decay. If an argument is an unsized array type, the array
+lvalue directly decays via array-to-pointer decay. \footnote{This results in
+\textit{input} parameters of unsized arrays being modifiable by a function.}
+
+\p Arguments to \textit{output} and \textit{input/output parameters} must be
+lvalues. \textit{Output parameters} are not initialized prior to the call; they
+are passed as an uninitialized cx-value (\ref{Basic.lval}). An \textit{output
+parameter} is only initialized explicitly inside the called function. It is
+undefined behavior to not explicitly initialize an \textit{output parameter}
+before returning from the function in which it is defined. The cx-value created
+from an argument to an \textit{input/output parameter} is initialized through
+copy-initialization from the lvalue argument expression. In both cases, the
+cx-value shall have the type of the parameter and the argument can be converted
+to that type through implicit or explicit conversion.
+
+\p If an argument to an \textit{output} or \textit{input/output parameter} is a
+constant sized array, the array is copied to a temporary cx-value following the
+same rules for any other data type. If an argument to an \textit{output} or
+\textit{input/output parameter} is an unsized array type, the array lvalue
+directly decays via array-to-pointer decay. An argument of a constant sized
+array of type \texttt{T[N]} can be converted to a cx-value of an unsized array
+of type \texttt{T[]} through array to pointer decay. An unsized array of type
+\texttt{T[]}, cannot be implicitly converted to a a constant sized array of type
+\texttt{T[N]}.
+
+\p On expiration of the cx-value, the value is assigned back to the argument
+lvalue expression following an inverted conversion if applicable. The argument
+expression must be of a type or able to convert to a type that has defined
+copy-initialization to and from the parameter type. The lifetime of the cx-value
+begins at argument expression evaluation, and ends after the function returns. A
+cx-value argument is passed by-address to the caller.
+
+\p If the lvalue passed to an \textit{output} or \textit{input/output parameter}
+does not alias any other parameter passed to that function, an implementation
+may avoid the creation of excess temporaries by passing the address of the
+lvalue instead of creating the cx-value.
+
+\p When a function is called parameters with object type must have completely
+defined type, and parameters of array of object must have completely defined
+element type.\footnote{HLSL \textit{output} and \textit{input/output parameters}
```

Comment:
Since we're talking about "parameters", should both of the "type"s in this sentence also be plural?